### PR TITLE
Make sure that eltyp is actually used for LinearAlgebra test

### PR DESCRIPTION
**Pull request by shirodkara**
without the parenthesis it would return variables of type Float64 on ppc64le .

### CODE REVIEW

**Comment by vchuravy** [comment]
Not just PPC specific, but we would eagerly promote the range to Float64.

File: stdlib/LinearAlgebra/test/lapack.jl [hunk 1]
```diff
@@ -196,7 +196,7 @@ end
 
 @testset "gebal/gebak" begin
     @testset for elty in (Float32, Float64, ComplexF32, ComplexF64)
-        A = rand(elty,10,10) * Diagonal(exp10.(range(-10, stop=10, length=10)))
+	A = rand(elty,10,10) * Diagonal(exp10.(range(elty(-10), stop=elty(10), length=10)))
```
Comment (vchuravy) [inline stdlib/LinearAlgebra/test/lapack.jl:199]:
```suggestion
        A = rand(elty,10,10) * Diagonal(exp10.(range(elty(-10), stop=elty(10), length=10)))
```

**Review by vchuravy — APPROVED** [review]

**Comment by ararslan** [comment]
This actually seems to break the relevant test:
```
Error in testset LinearAlgebra/lapack:
Test Failed at /buildworker/worker/tester_linux64/build/share/julia/stdlib/v1.6/LinearAlgebra/test/lapack.jl:205
  Expression: norm(diff(Avs ./ Bvs, dims = 1)) < 100 * eps(abs(float(one(elty))))
   Evaluated: NaN32 < 1.1920929f-5
```

**Comment by vchuravy** [comment]
@andreasnoack any suggestions for updating the tests to that it holds with `Float32`?

**Comment by andreasnoack** [comment]
The scaling is too extreme when using `Float32`. The scaling range should be a function of the element type so something like
```julia
A = rand(elty,10,10) * Diagonal(exp10.(range(log10(eps(real(elty)))/3*2, stop=-log10(eps(real(elty)))/3*2, length=10)))
```
is probably the right thing to do here.

File: stdlib/LinearAlgebra/test/lapack.jl [hunk 2]
```diff
@@ -196,7 +196,7 @@ end
 
 @testset "gebal/gebak" begin
     @testset for elty in (Float32, Float64, ComplexF32, ComplexF64)
-        A = rand(elty,10,10) * Diagonal(exp10.(range(-10, stop=10, length=10)))
+        A = rand(elty,10,10) * Diagonal(exp10.(range(elty(-10), stop=elty(10), length=10)))
```
Comment (vtjnash) [inline stdlib/LinearAlgebra/test/lapack.jl:199]:
```suggestion
        typescale = log10(eps(real(elty))) / 3 * 2
        A = rand(elty,10,10) * Diagonal(exp10.(range(typescale, stop=-typescale, length=10)))
```
(per @andreasnoack)

**Comment by vtjnash** [comment]
I'm confused, this apparently failed LinearAlgebra/dense on https://build.julialang.org/#builders/1/builds/2255, which isn't even something we changed here (@andreasnoack?):
```
Error in testset LinearAlgebra/dense:
Test Failed at /buildworker/worker/tester_linuxaarch64/build/share/julia/stdlib/v1.6/LinearAlgebra/test/dense.jl:343
  Expression: norm(z, p) == if p < 0
        0
    else
        Inf
    end
   Evaluated: NaN == Inf
Error in testset LinearAlgebra/dense:
Test Failed at /buildworker/worker/tester_linuxaarch64/build/share/julia/stdlib/v1.6/LinearAlgebra/test/dense.jl:343
  Expression: norm(z, p) == if p < 0
        0
    else
        Inf
    end
   Evaluated: NaN == Inf
```

**Comment by dkarrasch** [comment]
> I'm confused, this apparently failed LinearAlgebra/dense on https://build.julialang.org/#builders/1/builds/2255, which isn't even something we changed here (@andreasnoack?):
> 
> ```
> Error in testset LinearAlgebra/dense:
> Test Failed at /buildworker/worker/tester_linuxaarch64/build/share/julia/stdlib/v1.6/LinearAlgebra/test/dense.jl:343
>   Expression: norm(z, p) == if p < 0
>         0
>     else
>         Inf
>     end
>    Evaluated: NaN == Inf
> Error in testset LinearAlgebra/dense:
> Test Failed at /buildworker/worker/tester_linuxaarch64/build/share/julia/stdlib/v1.6/LinearAlgebra/test/dense.jl:343
>   Expression: norm(z, p) == if p < 0
>         0
>     else
>         Inf
>     end
>    Evaluated: NaN == Inf
> ```

I think that was an old issue that got fixed meanwhile. This will likely disappear after a rebase.

**Pull request merged**